core/node/state_keeper/src/seal_criteria/criteria/l1_l2_txs.rs

```diff
@@ -0,0 +1,96 @@
+use zksync_types::{
+    aggregated_operations::{L1_BATCH_EXECUTE_BASE_COST, L1_OPERATION_EXECUTE_COST},
+    ProtocolVersionId,
+};
+
+use crate::seal_criteria::{SealCriterion, SealData, SealResolution, StateKeeperConfig};
+
+#[derive(Debug)]
+pub(crate) struct L1L2TxsCriterion;
+
+impl SealCriterion for L1L2TxsCriterion {
+    fn should_seal(
+        &self,
+        config: &StateKeeperConfig,
+        _block_open_timestamp_ms: u128,
+        _tx_count: usize,
+        l1_tx_count: usize,
+        _block_data: &SealData,
+        _tx_data: &SealData,
+        _protocol_version_id: ProtocolVersionId,
+    ) -> SealResolution {
+        let block_l1_gas_bound =
+            (config.max_single_tx_gas as f64 * config.close_block_at_gas_percentage).round() as u32;
```

Comment:
it is actually probably better to change this to actually just check for the number of txs. For chains that settle on top of gateway we will have to set `max_single_tx_gas` to something big (billions of gas) and it may lead to having a large number of priority transactions. Not to say that gateway wont be able to handle a larger amount, but I am not sure how this discrepancy will play out (maybe the amount will become too large accidentally)